exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillOperatorTable.java

```diff
@@ -157,9 +159,23 @@ private void populateWrappedCalciteOperators() {
       } else if(calciteOperator instanceof SqlFunction) {
         wrapper = new DrillCalciteSqlFunctionWrapper((SqlFunction) calciteOperator,
             getFunctionListWithInference(calciteOperator.getName()));
+      } else if(calciteOperator instanceof SqlBetweenOperator) {
+        // During the procedure of converting to RexNode,
+        // StandardConvertletTable.convertBetween expects the SqlOperator to be a subclass of SqlBetweenOperator
+        final SqlBetweenOperator sqlBetweenOperator = (SqlBetweenOperator) calciteOperator;
+        wrapper = new SqlBetweenOperator(sqlBetweenOperator.flag, sqlBetweenOperator.isNegated()) {
+          @Override
+          public boolean checkOperandTypes(
```

Comment:
Correct. Otherwise, the regression reported in this DRILL-4525 will be failed by Calcite because Calcite does not allow between to be operated on timestamp and date.

By the way, this pull request is closed. The correct pull request is this one:
https://github.com/apache/drill/pull/439